tests/TestScripts/testSharding.py

```diff
@@ -0,0 +1,143 @@
+    if tags_arg:
+        cmd.append(tags_arg)
+    process = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    if stderr:
+        raise RuntimeError("Unexpected error output:\n" + process.stderr)
+
+    root = ET.fromstring(stdout)
+    result = [elem.attrib["name"] for elem in root.findall('./Group/TestCase')]
+
+    if len(result) < 2:
+        raise RuntimeError("Unexpectedly few tests listed (got {})".format(
+            len(result)))
+    return result
+
+def check_listed_tests_match(all_tests, sharded_tests):
+    # Make sure the number of tests in the full list, and the sharded lists are the same.
+    total_test_count = len(all_tests)
+    sharded_test_count = sum([len(shard) for shard in sharded_tests])
+
+    assert total_test_count == sharded_test_count, (
+        "Sharded test count ({}) does not match the total test count ({})".format(sharded_test_count, total_test_count)
+    )
+
+    # Make sure all the tests in the shards are from the full list, in the same order. Together with the previous check, this
+    # ensures that all tests in the full list, are in the shards without duplication.
+    test_index = 0
+    for shard_index, shard in enumerate(sharded_tests):
+        for shard_test_index, test_name in enumerate(shard):
+            assert test_name == all_tests[test_index], (
+                "Sharding does not split the test list while maintaining order {}:\n'{}' vs '{}'".format(test_index, test_name, all_tests[test_index])
+            )
+
+            test_index += 1
+
+def check_listed_and_executed_tests_match(listed_tests, executed_tests):
+    for shard_index, listed_shard in enumerate(listed_tests):
+        listed_shard_names = set(listed_shard)
+        executed_shard_names = set(executed_tests[shard_index])
+
+        listed_string = "\n".join(listed_shard_names)
+        exeucted_string = "\n".join(executed_shard_names)
+
+        assert listed_shard_names == executed_shard_names, (
+            "Executed tests do not match the listed tests:\nExecuted:\n{}\n\nListed:\n{}".format(exeucted_string, listed_string)
+        )
+
+def test_shards_cover_all_test(self_test_exe, test_case):
+    all_tests = list_tests(self_test_exe, test_case)
+    sharded_tests = [list_tests(self_test_exe, test_case, index) for index in range(test_case.shard_count)]
+
+    check_listed_tests_match(all_tests, sharded_tests)
+
+    executed_tests = [execute_tests(self_test_exe, test_case, index) for index in range(test_case.shard_count)]
+
+    check_listed_and_executed_tests_match(sharded_tests, executed_tests)
+
+
+def main():
+    self_test_exe, = sys.argv[1:]
+
+    # We want a random seed for the test, but want to avoid 0, because it has special meaning
+    shard_counts = [1, 5]
+    seeds = [random.randint(1, 2 ** 32 - 1), random.randint(1, 2 ** 32 - 1)]
+    tags = [["generators"], ["generators", "matchers"], []]
```

Comment:
Whichever tags end up being used, also add `~benchmarks`, in case some are added in the future for these tags.

Running even single benchmark adds significant runtime through measuring the execution environment.